src/Orleans/Threading/CancellationTokenWrapper.cs

```diff
@@ -0,0 +1,45 @@
+﻿using System;
+using System.Threading;
+
+namespace Orleans.Threading
+{
+    /// <summary>
+    /// Used as replacement of CancellationToken during network roundtrips
+    /// </summary>
+    [Serializable]
+    internal class CancellationTokenWrapper
```

Comment:
Wraps the cancellation token, so that on passing though serialization process linked `TaskCompletionSource` can be created. An alternative was to handle serialization of  `CancellationToken` directly in `BuiltInTypes`, but the thread that performs message handling and serialization have `RuntimeContext.Current`, which is essential for making grain calls, set to null.